common/arch/sdl/joy.cpp

```diff
@@ -466,5 +467,48 @@ int apply_deadzone(int value, int deadzone)
 		return 0;
 }
 
+bool joy_translate_menu_key(const d_event &event) {
+	if (event.type != EVENT_JOYSTICK_BUTTON_DOWN)
+		return false;
+#if DXX_MAX_BUTTONS_PER_JOYSTICK || DXX_MAX_HATS_PER_JOYSTICK || DXX_MAX_AXES_PER_JOYSTICK
+	// Find out which button has been pressed by decoding the button
+	// description text. This might not be the ideal way to do things,
+	// but we currently don't have data structures that allow us to do
+	// such a mapping in a saner way, so we'll take what we can get.
+	auto &e = static_cast<const d_event_joystickbutton &>(event);
+	Assert((e.button >= 0) && (e.button < joybutton_text.size()));
```

Comment:
`e.button` is `unsigned`, so it is always `>= 0`.  Some compiler configurations may warn about a comparison that can never fail.